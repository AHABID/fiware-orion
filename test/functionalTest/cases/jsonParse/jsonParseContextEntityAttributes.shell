# Make a registration using conv op /NGSI9/contextEntities/ENTITY_ID/attributes
echo "1: ++++++++++++++++++++"
url="/NGSI9/contextEntities/ENTITY_ID/attributes"
payload='{
  "metadatas" : [
    {
      "name" : "ID",
      "type" : "string",
      "value" : "1110"
    },
    {
      "name" : "cm2",
      "type" : "string",
      "value" : "XXX"
    }
  ],
  "duration" : "PT1H",
  "providingApplication" : "http://kz.tid.es/abc"
}'
curlJson ${url} "${payload}"

REG_ID=$(echo "$response" | grep -Po '(?<="registrationId" : ")[^"]*')

# Make a discovery to see that the registration went OK
echo "2: ++++++++++++++++++++"
url="/NGSI9/discoverContextAvailability"
payload='{
    "entities": [
        {
            "type": "",
            "isPattern": "false",
            "id": "ENTITY_ID"
        }
    ]
}'
curlJson ${url} "${payload}"

# Modify the first registration
echo "3: ++++++++++++++++++++"
url="/NGSI9/contextEntities/ENTITY_ID/attributes"
payload='{
  "metadatas" : [
    {
      "name" : "ID",
      "type" : "string",
      "value" : "1111"
    },
    {
      "name" : "cm2",
      "type" : "string",
      "value" : "YYY"
    }
  ],
  "duration" : "PT2H",
  "providingApplication" : "http://kz.tid.es/abc2",
  "registrationId" : "'$REG_ID'"
}'
curlJson ${url} "${payload}"

# Now discover that very registration again to see the change
echo "4: ++++++++++++++++++++"
url="/NGSI9/discoverContextAvailability"
payload='{
    "entities": [
        {
            "type": "",
            "isPattern": "false",
            "id": "ENTITY_ID"
        }
    ]
}'
curlJson ${url} "${payload}"

# Now try the GET method
echo "5: ++++++++++++++++++++"
curlJsonNoPayload "/NGSI9/contextEntities/ENTITY_ID/attributes"

# And a GET that fails
echo "6: ++++++++++++++++++++"
curlJsonNoPayload "/NGSI9/contextEntities/ENTITY_ID2/attributes"

# DELETE and PUT should give errors
echo "7: ++++++++++++++++++++"
curlJsonNoPayload "/NGSI9/contextEntities/ENTITY_ID/attributes" "-X DELETE"

echo "8: ++++++++++++++++++++"
url="/NGSI9/contextEntities/ENTITY_ID/attributes"
payload='{
  "metadatas" : [
    {
      "name" : "ID",
      "type" : "string",
      "value" : "1111"
    },
    {
      "name" : "cm2",
      "type" : "string",
      "value" : "YYY"
    }
  ],
  "duration" : "PT2H",
  "providingApplication" : "http://kz.tid.es/abc2"
}'
curlJson ${url} "${payload}" "-X PUT"

echo "9: ++++++++++++++++++++"

