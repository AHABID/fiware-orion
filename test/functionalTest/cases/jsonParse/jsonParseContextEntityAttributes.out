1: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: Tue, 03 Jun 2014 09:46:10 GMT

{
    "duration": "PT1H", 
    "registrationId": "538d9962c1caf828f4c08816"
}
2: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 302
Content-Type: application/json
Date: Tue, 03 Jun 2014 09:46:10 GMT

{
    "contextRegistrationResponses": [
        {
            "contextRegistration": {
                "entities": [
                    {
                        "id": "ENTITY_ID", 
                        "isPattern": "false", 
                        "type": ""
                    }
                ], 
                "providingApplication": "http://kz.tid.es/abc"
            }
        }
    ]
}
3: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: Tue, 03 Jun 2014 09:46:10 GMT

{
    "duration": "PT2H", 
    "registrationId": "538d9962c1caf828f4c08816"
}
4: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 303
Content-Type: application/json
Date: Tue, 03 Jun 2014 09:46:10 GMT

{
    "contextRegistrationResponses": [
        {
            "contextRegistration": {
                "entities": [
                    {
                        "id": "ENTITY_ID", 
                        "isPattern": "false", 
                        "type": ""
                    }
                ], 
                "providingApplication": "http://kz.tid.es/abc2"
            }
        }
    ]
}
5: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 303
Content-Type: application/json
Date: Tue, 03 Jun 2014 09:46:10 GMT

{
    "contextRegistrationResponses": [
        {
            "contextRegistration": {
                "entities": [
                    {
                        "id": "ENTITY_ID", 
                        "isPattern": "false", 
                        "type": ""
                    }
                ], 
                "providingApplication": "http://kz.tid.es/abc2"
            }
        }
    ]
}
6: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 94
Content-Type: application/json
Date: Tue, 03 Jun 2014 09:46:10 GMT

{
    "errorCode": {
        "code": "404", 
        "reasonPhrase": "No context element found"
    }
}
7: ++++++++++++++++++++
HTTP/1.1 405 Method Not Allowed
Content-Length: 0
Allow: POST, GET
Date: Tue, 03 Jun 2014 09:46:10 GMT

8: ++++++++++++++++++++
HTTP/1.1 405 Method Not Allowed
Content-Length: 0
Allow: POST, GET
Date: Tue, 03 Jun 2014 09:46:10 GMT

9: ++++++++++++++++++++
